packages/assets-controllers/src/NftController.test.ts

```diff
@@ -4471,6 +4471,47 @@ describe('NftController', () => {
   });
 
   describe('updateNftMetadata', () => {
+    it('should not update Nft metadata when preferences change and current and incoming state are the same', async () => {
+      const {
+        nftController,
+        triggerPreferencesStateChange,
+        triggerSelectedAccountChange,
+      } = setupController();
+      const spy = jest.spyOn(nftController, 'updateNftMetadata');
+      triggerSelectedAccountChange(OWNER_ACCOUNT);
+      // trigger preference change
+      triggerPreferencesStateChange({
+        ...getDefaultPreferencesState(),
+      });
+
+      expect(spy).toHaveBeenCalledTimes(0);
+    });
+    it('should call update Nft metadata when preferences change is triggered and at least ipfsGateway, openSeaEnabled or isIpfsGatewayEnabled change', async () => {
+      const {
+        nftController,
+        mockGetAccount,
+        triggerPreferencesStateChange,
+        triggerSelectedAccountChange,
+      } = setupController({
+        defaultSelectedAccount: OWNER_ACCOUNT,
+      });
+      const spy = jest.spyOn(nftController, 'updateNftMetadata');
+      const testNetworkClientId = 'mainnet';
+      mockGetAccount.mockReturnValue(OWNER_ACCOUNT);
+      await nftController.addNft('0xtest', '3', {
+        nftMetadata: { name: '', description: '', image: '', standard: '' },
+        networkClientId: testNetworkClientId,
+      });
+
+      triggerSelectedAccountChange(OWNER_ACCOUNT);
+      // trigger preference change
+      triggerPreferencesStateChange({
+        ...getDefaultPreferencesState(),
+        ipfsGateway: 'https://toto/ipfs/',
+      });
+
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
```

Comment:
nit: to match current test case formatting
```suggestion
    });

    it('should call update Nft metadata when preferences change is triggered and at least ipfsGateway, openSeaEnabled or isIpfsGatewayEnabled change', async () => {
      const {
        nftController,
        mockGetAccount,
        triggerPreferencesStateChange,
        triggerSelectedAccountChange,
      } = setupController({
        defaultSelectedAccount: OWNER_ACCOUNT,
      });
      const spy = jest.spyOn(nftController, 'updateNftMetadata');
      const testNetworkClientId = 'mainnet';
      mockGetAccount.mockReturnValue(OWNER_ACCOUNT);
      await nftController.addNft('0xtest', '3', {
        nftMetadata: { name: '', description: '', image: '', standard: '' },
        networkClientId: testNetworkClientId,
      });

      triggerSelectedAccountChange(OWNER_ACCOUNT);
      // trigger preference change
      triggerPreferencesStateChange({
        ...getDefaultPreferencesState(),
        ipfsGateway: 'https://toto/ipfs/',
      });

      expect(spy).toHaveBeenCalledTimes(1);
    });

```